### PR TITLE
Development docker

### PR DESCRIPTION
**Pull request by BastekBielawski**
# Description

- Build in CRON
- Worker based on Supervisor
- Composer v1 support

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

# Checklist:
- [x] I have read the contribution requirements and fulfill them.
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: .env.dist [hunk 1]
```diff
@@ -1,76 +1,9 @@
...
-IMPORT_WORKER_SLEEP=30
-EXPORT_WORKER_SLEEP=30
-WORKER_MEMORY_LIMIT=128M
-IMPORT_WORKER_MEMORY_LIMIT=512M
-EXPORT_WORKER_MEMORY_LIMIT=512M
-WORKER_TIME_LIMIT=3600
-###< Worker ###
+###> PHP container ###
+# For Windows and Mac OS X users
+PHP_IDE_CONFIG=host.docker.internal
```
Comment (piotrkreft) [inline .env.dist:3]: Fairly sure this should be as for Linux version `serverName=***`

File: config/php/docker-entrypoint.sh [hunk 2]
```diff
@@ -62,148 +18,37 @@ function waitUntil() {
...
-    bin/console messenger:setup-transports --no-interaction segment
-    bin/console messenger:setup-transports --no-interaction completeness
-    bin/console messenger:setup-transports --no-interaction notification
+>&2 echo "Generate JWT keys"
+genereJwtKeys "config/jwt/private.pem" "config/jwt/public.pem"
 
-    >&2 echo "messenger initialization finished"
-fi
+>&2 echo "Composer install"
+#composer install --prefer-dist --no-progress --no-suggest --no-interaction
```
Comment (piotrkreft) [inline config/php/docker-entrypoint.sh]: I guess it could stay. Otherwise will have to be done manually

File: docker-compose.yml [hunk 3]
```diff
@@ -1,195 +1,81 @@
...
-            APP_USER: ${APP_USER:-ergonode}
-            APP_USER_PASSWORD: ${APP_USER_PASSWORD:-123}
-            APP_DB: ${APP_DB:-ergonode}
-            APP_TEST_DB: ${APP_TEST_DB:-ergonode_test}
+            POSTGRES_USER: "postgres"
+            POSTGRES_PASSWORD: "123"
+            POSTGRES_DB: "postgres"
         ports:
-            - "${EXPOSED_POSTGRES_PORT:-5432}:5432"
+            - "5432"
```
Comment (piotrkreft) [inline docker-compose.yml]: Could we leave at least ports configurable? It's easier to set up multiple applications if needed.

File: docker-compose.yml [hunk 4]
```diff
@@ -1,195 +1,81 @@
...
-        working_dir: /srv/app
-        image: ${CONTAINER_REGISTRY_BASE:-ergonode}/php:${IMAGE_TAG:-latest}
-        volumes:
-            - ./backend:/srv/app:delegated
-        environment:
-            <<: *php-environment
-        command: ["crond"]
+            - "postgres"
+            - "mailhog"
+            - "rabbitmq"
```
Comment (piotrkreft) [inline docker-compose.yml]: You can have synchronous local configuration and don't want to raise extra containers

**Review by piotrkreft — CHANGES_REQUESTED** [review]

File: config/php/docker-entrypoint.sh [hunk 5]
```diff
@@ -62,148 +18,37 @@ function waitUntil() {
...
-    su -s /bin/bash www-data -c 'crontab -l'
-    su -s /bin/bash www-data -c "touch /srv/app/var/log/crond.log"
-    crond -f &
-    tail -f /srv/app/var/log/crond.log
-
-else
-  exec docker-php-entrypoint "$@"
-fi
+>&2 echo "Database migrations"
+bin/console ergonode:migrations:migrate --no-interaction --allow-no-migration
```
Comment (piotrkreft) [inline config/php/docker-entrypoint.sh:46]: Perhaps create DB as well if not exists

File: docker-compose.yml (hunk 4 above)
Comment (BastekBielawski) [reply docker-compose.yml]: Sure, no problem

File: docker-compose.yml (hunk 3 above)
Comment (BastekBielawski) [reply docker-compose.yml]: Ok

File: config/php/docker-entrypoint.sh (hunk 2 above)
Comment (BastekBielawski) [reply config/php/docker-entrypoint.sh]: It will be uncomment when i finish :)

File: config/php/docker-entrypoint.sh (hunk 5 above)
Comment (BastekBielawski) [reply config/php/docker-entrypoint.sh:46]: Database creation it's done in `./config/postgres/init.sql`

File: .env.dist (hunk 1 above)
Comment (BastekBielawski) [reply .env.dist:3]: When i finish we can check because i find it on official XDebug site

**Comment by BastekBielawski** [comment]
I need to contact with frontend developers to establish nuxtjs

**Pull request closed without merging**